structure/structure.py

```diff
@@ -15,24 +15,52 @@
 from griptape.tools import Calculator
 
 
-# Returns True if this program is being run from within Griptape Cloud or via the Skatepark emulator
-def is_running_in_managed_environment():
+def is_running_in_managed_environment() -> bool:
+    """Determine if the program is running in a managed environment (e.g., Griptape Cloud or Skatepark emulator).
+    Returns:
+        bool: True if the program is running in a managed environment, False otherwise.
+    """
     return "GT_CLOUD_RUN_ID" in os.environ
 
 
+def get_listener_url() -> str:
+    """The event driver requires a URL, stored in the GT_CLOUD_BASE_URL 
+    environment variable.
+    In Griptape Cloud, this environment variable is provided for you.
+    In Skatepark, you may optionally provide your own if you have overridden
+    the default URL and port that the emulator is listening on. Otherwise,
+    it will use the default URL and port.
+
+    Returns:
+        The URL of the host listening to the Structure.
+    """
+    return os.environ.get("GT_CLOUD_BASE_URL")
+
+
+def get_listener_api_key() -> str:
+    """The event driver expects a Griptape API Key as a parameter.
+    When your program is running in Griptape Cloud, you will need to provide a 
+    valid Griptape API Key in the GT_CLOUD_API_KEY environment variable 
+    in order to authorize calls. 
+    You can create an API Key by visiting https://cloud.griptape.ai/keys 
+    When running in Skatepark, the API key is not needed since it isn't validating calls.
+
+    Returns:
+        The Griptape API Key to authorize calls.
+    """
+    return os.environ.get("GT_CLOUD_API_KEY")
```

Comment:
This is where the the warning should go, yes?

Reply:
I thought we could put the warning in Skatepark itself, right? If you'd prefer it here, I can do it here.

Reply:
Oh I see. We can add it to Skatepark but the user may not be looking at the server logs.

Reply:
I'll put a warning here.